modules/apps/segments/segments-service/src/main/java/com/liferay/segments/service/impl/SegmentsExperienceLocalServiceImpl.java

```diff
@@ -494,6 +484,143 @@ private long _getPublishedLayoutClassPK(long classPK) {
 		return classPK;
 	}
 
+	private List<SegmentsExperience> _getSegmentExperiences(
+		SegmentsExperience segmentsExperience) {
+
+		List<SegmentsExperience> aboveSegmentsExperiences = new ArrayList<>(
+			segmentsExperiencePersistence.findByG_C_C_GtP(
+				segmentsExperience.getGroupId(),
+				segmentsExperience.getClassNameId(),
+				segmentsExperience.getClassPK(),
+				segmentsExperience.getPriority()));
+
+		Collections.reverse(aboveSegmentsExperiences);
+
+		List<SegmentsExperience> belowSegmentsExperiences =
+			segmentsExperiencePersistence.findByG_C_C_LtP(
+				segmentsExperience.getGroupId(),
+				segmentsExperience.getClassNameId(),
+				segmentsExperience.getClassPK(),
+				segmentsExperience.getPriority());
+
+		return Stream.concat(
+			Stream.concat(
+				aboveSegmentsExperiences.stream(),
+				Stream.of(segmentsExperience)),
+			belowSegmentsExperiences.stream()
+		).collect(
+			Collectors.toList()
+		);
+	}
+
+	private SegmentsExperience _swapAllListSegmentsExperience(
```

Comment:
I'm not in love with this name, not sure how it should be named though. `_swapSegmentsExperiences`? (plural)